### PR TITLE
dts: microchip: add support for MEC1727 dts

### PR DESCRIPTION
**Pull request by jvasanth1**
MEC1727 has internal flash, so add the definition in
a separate MEC1727nsz.dtsi file.

Signed-off-by: Jay Vasanth <jay.vasanth@microchip.com>

### CODE REVIEW

**Comment by yperess** [comment]
LGTM, but someone else should approve.

**Review by yperess — DISMISSED** [review]
Played around with this PR a bit and it seems to work from what I can tell.

**Comment by jvasanth1** [comment]
@keith-zephyr can you help to review and approve this PR?

**Review by carlescufi — DISMISSED** [review]

File: dts/arm/microchip/mec1727nsz.dtsi [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+		      &gpspi_clk_gpio117
+		      &gpspi_io0_gpio074
+		      &gpspi_io1_gpio075
+		      &gpspi_wp_n_gpio076 >;
+	pinctrl-names = "default";
+
+	int_flash: sst25pf040@0 {
+		compatible ="jedec,spi-nor";
+		/*  4194304 bits = 512K Bytes */
+		size = <0x400000>;
```
Comment (fabiobaltieri) [inline dts/arm/microchip/mec1727nsz.dtsi:40]: how about using DT_SIZE_M(4)? (and updating the comment specifying that this is 4Mbit

File: dts/arm/microchip/mec1727nsz.dtsi [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+		label = "SST25PF040";
+		reg = <0>;
+		spi-max-frequency = <40000000>;
```
Comment (fabiobaltieri) [inline dts/arm/microchip/mec1727nsz.dtsi:43]: Similarly, we could use `DT_FREQ_M(40)` here.

File: dts/arm/microchip/mec1727nsz.dtsi (hunk 1 above)
Comment (jvasanth1) [reply dts/arm/microchip/mec1727nsz.dtsi:40]: I tried but I get `parse error: malformed value`
do you have an example of the usage?

File: dts/arm/microchip/mec1727nsz.dtsi (hunk 2 above)
Comment (jvasanth1) [reply dts/arm/microchip/mec1727nsz.dtsi:43]: I tried but I get `parse error: malformed value`
do you have an example of the usage?

Comment (fabiobaltieri) [reply dts/arm/microchip/mec1727nsz.dtsi:43]: That is odd, they are used in load of places (`dts/arm/nuvoton/npcx9.dtsi` for example), did you add
`#include <freq.h>`? (and `mem.h` for `DT_SIZE_M`. It's a normal C preprocessor, shouldn't do anything special.

**Review by fabiobaltieri — CHANGES_REQUESTED** [review]
(marking as change requested so it falls off the filter while we work on the freq/mem comment)

File: dts/arm/microchip/mec1727nsz.dtsi (hunk 1 above)
Comment (jvasanth1) [reply dts/arm/microchip/mec1727nsz.dtsi]: fixed

File: dts/arm/microchip/mec1727nsz.dtsi (hunk 2 above)
Comment (fabiobaltieri) [reply dts/arm/microchip/mec1727nsz.dtsi]: Thanks, one more note, how about adding those two includes in `mec172xnsz.dtsi` instead? that way they are available for all sub-families.

Comment (jvasanth1) [reply dts/arm/microchip/mec1727nsz.dtsi]: moved the two includes to mec172xnsz.dtsi

**Review by fabiobaltieri — APPROVED** [review]
Thanks!!

**Review by carlescufi — APPROVED** [review]

**Pull request merged**